### PR TITLE
feat(TimelineEvent): add timeline component

### LINKED ISSUE #1131: [Timeline] add visual timeline component
### Short Description

FedNow has this feature built ad-hoc, but we'll want to reuse it other places.

TK

### Mockup

[figma](https://www.figma.com/file/TG6rH21v231JFcZy9jaIz6/fednow-manager?type=design&node-id=1720-13977&mode=design&t=58B42hUk2RWUdvLl-0)

### Responsibilities

**This component should be responsible for the following:**
- Managed children (`.Item` components)
- Status type
- Thru line style based on status
- Managing layout between items
- Taking an icon or image

**This component should NOT be responsible for...**
- JSX within the right hand of the media block. Let consumers pass that into `Item` for flexibility

### Props Signature

```
<Timeline {...top level config props if applicable}>
    <Timeline.Item kind="pending">Any JSX</Timeline.Item>
    <Timeline.Item icon="user">Any JSX</Timeline.Item>
    <Timeline.Item kind="start">Any JSX</Timeline.Item>
</Timeline>
```

- [ ] ❓ do we really need the manager? Why not just `TimelineNode`? Try that first.

### Layout

Use an SVG line designed to overflow by node size * 0.5

Can probably use grid to lock in the spacing

### PR DESCRIPTION
**Pull request by akdetrick**
closes #1131 

Adds `TimelineEvent` component we can use to build timelines with visually connected nodes.

### CODE REVIEW

**Comment by akdetrick** [comment]
Awaiting design approval. Some changes are expected before we mark ready to review.

File: src/TimelineEvent/index.scss [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  flex-grow: 1;
+}
+
+.nds-timeline-line {
+  position: absolute;
+  z-index: 0;
+  left: 0;
+  right: 0;
+  // height of event + gap
+  height: calc(100% + var(--space-l));
```
Comment (akdetrick) [inline src/TimelineEvent/index.scss:33]: This matches the height of the SVG to the height of the root element in `TimelineEvent`, plus the height of a node, creating the overflap

File: src/TimelineEvent/index.scss [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  }
+}
+
+.nds-timeline-event--pending {
+  .nds-timeline-node {
+    animation: flash 1.5s alternate infinite;
+  }
+  svg line {
+    stroke-dasharray: 2;
+  }
```
Comment (akdetrick) [inline src/TimelineEvent/index.scss:72]: When a node is pending, we set this property on the SVG line so it's drawn with dashes

File: src/TimelineEvent/index.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import PropTypes from "prop-types";
+import iconSelection from "src/icons/selection.json";
+import cc from "classcat";
+
+export const VALID_ICON_NAMES = iconSelection.icons.map(
+  (icon) => icon.properties.name
+);
+
+const TimelineEvent = ({ kind = "node", icon, imgUrl, initial, children }) => {
```
[the remark this answers is not in the document]
Comment (akdetrick) [reply src/TimelineEvent/index.js:10]: Yes, "node" is a past event.
I treated this as "Everything not pending is past".

We can always add additional states as new `kind` values later, but for now I think this makes sense as a default

[the remark this answers is not in the document]
Comment (laurenharpole) [reply src/TimelineEvent/index.js:10]: sounds good to me!

**Review by laurenharpole — APPROVED** [review]
Super excited about this looks great

**Review by aundus — APPROVED** [review]
Very excited about this

**Pull request merged**